src/main/java/com/mesabrook/ib/blocks/gui/telecom/GuiBootScreen.java

```diff
@@ -0,0 +1,117 @@
+import com.mesabrook.ib.util.handlers.PacketHandler;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.gui.GuiButton;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumHand;
+
+import java.io.IOException;
+
+
+public class GuiBootScreen extends GuiPhoneBase
+{
+    private int timerToNextScreen = 0;
+    private int fadeAnimationTimer = 0;
+    private String currentTexture;
+
+    public GuiBootScreen(ItemStack phoneStack, EnumHand hand)
+    {
+        super(phoneStack, hand);
+        fadeAnimationTimer++;
+    }
+
+    @Override
+    protected String getInnerTextureFileName()
+    {
+        if(fadeAnimationTimer == 0)
+        {
+            currentTexture = "app_screen_no_bar.png";
+        }
+        if(fadeAnimationTimer == 50)
+        {
+            currentTexture = "app_screen_boot_0.png";
+        }
+        if(fadeAnimationTimer == 100)
+        {
+            currentTexture = "app_screen_boot_1.png";
+        }
+        if(fadeAnimationTimer == 150)
+        {
+            currentTexture = "app_screen_boot_2.png";
+        }
+        if(fadeAnimationTimer == 200)
+        {
+            currentTexture = "app_screen_boot_3.png";
+        }
+        if(fadeAnimationTimer == 250)
+        {
+            currentTexture = "app_screen_boot_4.png";
+        }
+        if(fadeAnimationTimer == 300)
+        {
+            currentTexture = "app_screen_boot_5.png";
+        }
+
+        return currentTexture;
+    }
+
+    @Override
+    protected boolean renderControlBar() {
+        return false;
+    }
+
+    @Override
+    protected boolean renderTopBar() {
+        return false;
+    }
+
+    @Override
+    public void initGui()
+    {
+        super.initGui();
+
+        ClientSoundPacket soundPacket = new ClientSoundPacket();
+        soundPacket.pos = Minecraft.getMinecraft().player.getPosition();
+        soundPacket.soundName = "minedroid_firstboot";
+        PacketHandler.INSTANCE.sendToServer(soundPacket);
+    }
+
+    @Override
+    protected void doDraw(int mouseX, int mouseY, float partialticks)
+    {
+        super.doDraw(mouseX, mouseY, partialticks);
+        timerToNextScreen++;
+        fadeAnimationTimer++;
+
+        if(timerToNextScreen >= 1500)
+        {
+            finishBoot();
+        }
+    }
+
+    private void finishBoot()
+    {
+        ClientSoundPacket soundPacket = new ClientSoundPacket();
+        soundPacket.pos = Minecraft.getMinecraft().player.getPosition();
+        soundPacket.soundName = "minedroid_startup";
+        PacketHandler.INSTANCE.sendToServer(soundPacket);
+        GuiLockScreen lock = new GuiLockScreen(Minecraft.getMinecraft().player.getHeldItem(hand), hand);
+        Minecraft.getMinecraft().displayGuiScreen(lock);
+    }
+
+    @Override
+    protected void actionPerformed(GuiButton button) throws IOException
+    {
+        super.actionPerformed(button);
+    }
+
+    @Override
+    public void onGuiClosed()
```

Comment:
I don't think you need to override this because those numbers are specific to this screen. Next time it shows up, those will be back to 0. That would be a different story if `timerToNextScreen` and `fadeAnimationTimer` were static.

Reply:
See above lol

Reply:
Though there is an interesting bug where sometimes both timers would tick too fast. I think that's a broader Minecraft problem though, could be wrong lmao.

Reply:
> See above lol

I actually did take a look around and didn't see anything



> Though there is an interesting bug where sometimes both timers would tick too fast. I think that's a broader Minecraft problem though, could be wrong lmao.

Yeah, this is because the timers are based on your current FPS, not the rate in which the server is ticking. Fast FPS means you will get through the screen faster, very slow FPS means that these screens will last _much_ longer. Not a whole lot we can do about that, though, except checking world time, and then feels overkill.

Reply:
![spin-fish](https://github.com/Mesabrook-Development-Team/Immersibrook/assets/1634865/4f1db2d4-4d0e-4bbf-9c11-dc370f66bc64)

Agreed